### PR TITLE
[SPARK-7535.1] [MLLIB] minor changes to the pipeline API

### PR DESCRIPTION
**Pull request by mengxr**
1. removed `Params.validateParams(extra)`
2. added `Evaluate.evaluate(dataset, paramPairs*)`
3. updated `RegressionEvaluator` doc

@jkbradley

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala [hunk 1]
```diff
@@ -37,8 +37,8 @@ final class RegressionEvaluator(override val uid: String)
   def this() = this(Identifiable.randomUID("regEval"))
 
   /**
-   * param for metric name in evaluation
-   * @group param supports mse, rmse, r2, mae as valid metric names.
+   * param for metric name in evaluation (supports rmse, mse, r2, and mae)
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala]: State default in Scala doc

File: mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala [hunk 2]
```diff
@@ -102,12 +102,6 @@ class CrossValidator(override val uid: String) extends Estimator[CrossValidatorM
   /** @group setParam */
   def setNumFolds(value: Int): this.type = set(numFolds, value)
 
-  override def validateParams(paramMap: ParamMap): Unit = {
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala]: Isn't this still needed?

**Comment by jkbradley** [comment]
Looks good other than those 2 items

File: mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala (hunk 2 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala]: No, `validateParams(extra)` is removed. We try to limit the places where users can specify extra params, which helps simplify the implementation.

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala]: done

File: mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala (hunk 2 above)
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala]: Sorry, I should have been clearer: CrossValidator still needs to call validateParams() for each of the estimatorParamMaps.  It should copy the estimator with each paramMap, and then call validateParams for that copy.

**Comment by jkbradley** [comment]
LGTM pending tests!

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
Merged into master and branch-1.4.

**Pull request closed without merging**